app.rb

```diff
@@ -35,11 +35,9 @@ def create_person
     }
 
     chosen_option = choose_from(options)
-
-    print 'Name: '
-    name = gets.chomp
-    print 'Age: '
-    age = gets.chomp
+    
+    name = getInput('Name:')
+    age = getInput('Age:')
 
     the_person = chosen_option == '1' ? create_student(name, age) : create_teacher(name, age)
```

Comment:
- [ ] Good job so far 👍 but all requests for input and input validation should've been placed in the `main.rb` file or a different file whose responsibilities are to request or validate input. Kindly check this [lesson](`https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid`) for more information on that. All the methods in your `app.rb` should be like your `list_all_books` or `list_all_people` methods which just run the code without asking for any input or validation.

Reply:
Hi @Kingstalux. Thanks for your review.

First of all, `main.rb` in our project is only responsible for running the app. May you please give me some good resources on how to follow OOP and SOLID for a project like this? I just checked your repo and noticed neither `main` nor `app` are class.

I can simply make another class to pass this task but I am not sure if this is what Microverse wants.

Our idea for app was this: App class has one responsibility for getting inputs and accordingly call relevant entity to do the task.

Reply:
Hello @ahangarha,
- If you can put all the input requests and validations in another class it would be great since that will clean up the `app.rb` file. 
- Also in my repo I did not do this project since I worked with my partner's repo.